### DISCUSSION TITLE
Lock/Hide Export

### DISCUSSION THREAD

**Discussion by NoobRob** [sumatrapdfreader/sumatrapdf#4152] [Ideas]
it would be great if you could use command-line (or something else) to prevent all output, such as printing, from being displayed. this would prevent users from printing everything. It should also be prevented that you can save PDFs separately or send them by e-mail.

So actually suppress EVERYTHING except display functions

**Comment by GitHubRulesOK** [comment]
NOT possible Adobe wrote the OpenLicensed PDF specification. Which is now an International Standard based on Documents as files and all files are Word Processing / System runtime editor input and output. (one reason you dont want them to have internal scripts.)
This one is non readable unless I give you a password thus  you should not be able to print it unless you crack the encryption and open it then it is viewable thus printable. https://github.com/GitHubRulesOK/MyNotes/blob/master/AppNotes/protected.pdf

You will need to use Adobe software to restrict users by only open in Acrobat Readers with all their printers options set to "limited"

So you can edit any file once it is viewable. There are 2 cases where a PDF is not edit or print worthy in a PDF Reader 
- One is where you forgot the password to open that personal data (so did not give any body else nor get the password). 
- The other is it is corrupt beyond redemption for analysis, thus not worth viewing.  

There is a third case where the Shared Password to OPEN is "desirable" and that is Forms Submittal, since without a password to transmit the encrypted file, it is open pickings to any eavesdropper.  But SumatraPDF does not RUN submittals.

**Comment by NoobRob** [comment]
Thank you very much for your quick reply.

I understand the specifications, but I think they are insufficient in terms of data protection.
The internal pdf protection does not protect against copying and passing on the password. A user can therefore copy all files and pass on the passwords to third parties.


The requirement results from the following situation: We have over 5000 PDFs (each 500MB - 1200MB). These are the former paper folders that have been scanned. We need to protect this data so that it cannot be copied or printed. DMS is not an option, because the internal DMS-Document-Viewer doesnt support big files.

I have written a program that displays the PDFs as a list. The user clicks on one and a viewer opens this file. The user himself has no physical access to the data. Copying is therefore not possible.


I currently use a plugin to display the PDFs, which does not allow printing and separate saving. The problem here, however, is that browsing and opening takes quite a long time. Sumatra is considerably faster.

It is not about PDF specification. But it would be nice if users just had a simple interface in which they could only search ;)

**Comment by GitHubRulesOK** [comment]
SumatraPDF is designed to Assist users with high speed access to the PDF data so several oddities spring to mind such as

Ideally PDF was designed to be small files of KB not MB
so not more than 500 scanned images or say 100 MB (if each was 200 KB at worst )
images as single highly compressed objects are naturally way slower than copy able text contents.
However stored as JPG.Zip (CBZ) format may be much smaller yet fast to decompress and less easy to copy.

As soon as you add accessible (Human compatible text, it is searchable, thus available to be extracted as Audio Enabled or clipboard accessible) Unfortunately SumatraPDF does not expose the text as Printer OR Audio compatible, but windows allows text to be copied via the clipboard as it is a WINDOWS feature. Likewise it is Windows provides file handling such as file seek or duplicate or save etc.

**Comment by kjk** [comment]
See configuring in restricted mode: https://www.sumatrapdfreader.org/docs/Configure-for-restricted-use

You can disable any functionality that requires disk access, internet access, printing etc.

https://github.com/sumatrapdfreader/sumatrapdf/blob/master/docs/sumatrapdfrestrict.ini
